### PR TITLE
SHEL-75-Register-Page

### PR DESCRIPTION
**Pull request by KwinyarutP**

### CODE REVIEW

**Comment by KwinyarutP** [comment]
@wit03 review pls

File: src/app/(auth)/_component/FoundationContent.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      {children}
+    </Button>
+  );
+};
+
+export default function FoundationContent(){
+
+  const [form] = Form.useForm();
+  const router = useRouter(); //navigate to success
+  const [fileList, setFileList] = useState([]); //store file that user upload
```
Comment (TGRZiminiar) [inline src/app/(auth)/_component/FoundationContent.tsx:38]: Add type to this use state

File: src/app/(auth)/_component/FoundationContent.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    setFileList(fileList); // Update fileList state
+  }; 
+
+  return (
+    <> 
+      <Form
+        form={form}
+        className="mt-10 flex-col"
+        onFinish={(values) => {
+            console.log("Form submitted:", values);
```
Comment (TGRZiminiar) [inline src/app/(auth)/_component/FoundationContent.tsx:50]: remove console log

File: src/app/(auth)/_component/RegisterContent.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+import type { FormInstance } from "antd";
+import Link from "next/link";
+import { useRouter } from "next/navigation";
+
+
+interface SubmitButtonProps {
+  form: FormInstance;
+}
+export const SubmitButton: React.FC<React.PropsWithChildren<SubmitButtonProps>> = ({form,children,}) => {
+  const [submittable, setSubmittable] = useState(false);
```
Comment (TGRZiminiar) [inline src/app/(auth)/_component/RegisterContent.tsx:12]: Give a type to state

File: src/app/(auth)/_component/RegisterContent.tsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  const [form] = Form.useForm();
+  const router = useRouter(); //navigate to success
+
+return (
+    <> 
+          <Form
+            form={form}
+            className="mt-10 flex-col"
+            onFinish={(values) => {
+                console.log("Form submitted:", values);
```
Comment (TGRZiminiar) [inline src/app/(auth)/_component/RegisterContent.tsx:45]: remove console.log

File: src/app/(auth)/register/page.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
+"use client";
```
Comment (TGRZiminiar) [inline src/app/(auth)/register/page.tsx:1]: move page.tsx into CPage.tsx look how other people do it
Vercel can't deploy if page.tsx is clientside

File: src/app/(auth)/registersuccess/page.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
+"use client"
```
Comment (TGRZiminiar) [inline src/app/(auth)/registersuccess/page.tsx:1]: move page.tsx into CPage.tsx look how other people do it
Vercel can't deploy if page.tsx is clientside

**Review by TGRZiminiar — CHANGES_REQUESTED** [review]

**Pull request merged**